src/source/tile_cache.js

```diff
@@ -12,14 +12,14 @@ import type Tile from './tile';
  */
 class TileCache {
     max: number;
-    data: {[key: number | string]: Array<{ value: Tile, timeout: ?TimeoutID}>};
+    data: {[key: number | string]: Array<{ value: any, timeout: ?TimeoutID}>};
```

Comment:
Why do these types need to be relaxed? If it is to allow usage of methods only available on sub-classes of `Tile`, maybe those should be added to `Tile` or a type check used to conditionally execute those methods.

Reply:
Because this cache never invokes anything on the `Tile` type, the design already uses an `onRemove` callback to define what needs to be called on removal of an element. So we can use it as datastore to store any data that has a 1:1 OR 1:n relationship with a tile, and automatically guarantee an upper bound on the memory consumption of those resources.

I want to use it for storing and caching `Texture` references in `Painter`, instead of adding more properties to `Tile`.